kernel/src/scan/mod.rs

```diff
@@ -103,9 +108,63 @@ pub struct ScanResult {
 
 /// Scan uses this to set up what kinds of columns it is scanning. A reference to the field is
 /// stored so we can get the name and data_type out when building the read expression
-enum ColumnType<'a> {
-    Selected(&'a StructField),
-    Partition(&'a StructField),
+pub enum ColumnType<'a> {
+    Selected(Cow<'a, StructField>),
+    Partition(Cow<'a, StructField>),
+}
+
+impl<'a> ColumnType<'a> {
+    pub(crate) fn into_owned(self) -> ColumnType<'static> {
+        match self {
+            ColumnType::Selected(field) => {
+                let field = field.into_owned();
+                ColumnType::Selected(Cow::Owned(field))
+            }
+            ColumnType::Partition(field) => {
+                let field = field.into_owned();
+                ColumnType::Partition(Cow::Owned(field))
+            }
+        }
+    }
+}
+/// One file to scan, plus associated metadata
+pub struct ScanFile {
```

Comment:
this struct will probably go away once we settle on how we want to use engine_data